crates/pathfinder/src/state/sync/l2.rs

```diff
@@ -185,7 +187,21 @@ async fn download_block(
     let result = sequencer.block(block_number.into()).await;
 
     match result {
-        Ok(block) => Ok(DownloadBlock::Block(Box::new(block))),
+        Ok(block) => {
+            let block = Box::new(block);
+            // Check if block hash is correct.
+            let expected_block_hash = block.block_hash.unwrap();
+            let verify_hash = tokio::task::spawn_blocking(move || -> anyhow::Result<_> {
+                let verify_result = verify_block_hash(&block, chain, expected_block_hash)?;
```

Comment:
```suggestion
                let verify_result = verify_block_hash(&block, chain, expected_block_hash).with_context(|| format!("Verify block {}", block_number))?
```

Perhaps this would work with my other comment?